src/output-json-stats.c

```diff
@@ -265,23 +265,33 @@ json_t *StatsToJSON(const StatsTable *st, uint8_t flags)
         uint32_t x;
         for (x = 0; x < st->ntstats; x++) {
             uint32_t offset = x * st->nstats;
-
-            // Stats for for this thread.
-            json_t *thread = json_object();
-            if (unlikely(thread == NULL)) {
-                json_decref(js_stats);
-                json_decref(threads);
-                return NULL;
-            }
+            const char *tm_name = NULL;
+            json_t *thread = NULL;
 
             /* for each counter */
             for (u = offset; u < (offset + st->nstats); u++) {
                 if (st->tstats[u].name == NULL)
                     continue;
 
-                // Seems this holds, but assert in debug builds.
-                DEBUG_VALIDATE_BUG_ON(
-                        strcmp(st->tstats[offset].tm_name, st->tstats[u].tm_name) != 0);
+                if (st->tstats[u].tm_name == NULL) {
+                    SCLogWarning("stats entry without tm_name name=%s", st->tstats[u].name);
```

Comment:
I missed this before ... we'll generate a log message each time the stats are collected when this occurs.

Have you witnessed this occurring?

Does it indicate a severe error? Should Suricata stop in this case? I don't know if a warning message is the right step here and perhaps the `DEBUG_VALIDATE_ON` should be retained?

Reply:
> Have you witnessed this occurring?

I have not, will sswitch to DEBUG_VALIDATE_ON